### PR TITLE
VIP Scores plot

### PR DESCRIPTION
**Pull request by dchiquit**
* Add `VipPlot.vue`
* Add VipPlot to OPLSDA and PLSDA pages
  * PLSDA page has a complete set of VIP scores for each "numComponent", so there is also a selector to choose which component to render
  * Both have a toggle to sort based on component name or on VIP score

### CODE REVIEW

File: web/src/components/vis/VipPlot.vue [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+              tooltip.transition()
+                .duration(duration)
+                .style('opacity', 0.0);
+            }))
+          .transition()
+          .duration(fadeInDuration)
+          .attr('x', (d) => scaleX.value(d.col))
+          .attr('y', (d) => scaleY.value(d.vip))
+          .attr('width', scaleX.value.bandwidth())
+          .attr('height', (d) => dheight.value - scaleY.value(d.vip));
```
Comment (waxlamp) [inline web/src/components/vis/VipPlot.vue:117]: It looks like these same values are set on the enter selection, which makes it seem like these transitions aren't really needed. Do they trigger in some other situation?

**Review by waxlamp — APPROVED** [review]
Just had one question about transitions, but it shouldn't block merging of this PR.

File: web/src/components/vis/VipPlot.vue (hunk 1 above)
Comment (dchiquit) [reply web/src/components/vis/VipPlot.vue:117]: This is what happened when I removed height:
image
You are correct that width is superfluous, as it doesn't change once the plot is loaded.

Comment (dchiquit) [reply web/src/components/vis/VipPlot.vue:117]: The transition happens when toggling the `Sort VIP plot values` toggle, so yes the transition is required. It's pretty satisfying to watch.

Comment (waxlamp) [reply web/src/components/vis/VipPlot.vue]: :+1:

**Review by subdavis — APPROVED** [review]
Tested, works great.

**Review by subdavis — APPROVED** [review]

**Pull request merged**